### ISSUE TITLE
Pickle checkpoints fail to restore a state of the DipolarP3M

### ISSUE THREAD

**Issue by Vladimir-Z** [espressomd/espresso#4088]
When we trying to save and load object of magnetostatics.DipolarP3M  we got the following error
```
    File "actors.pyx", line 203, in espressomd.actors.Actors.__setstate__
    File "actors.pyx", line 70, in espressomd.actors.Actor._activate
    File "magnetostatics.pyx", line 188, in espressomd.magnetostatics.DipolarP3M._activate_method
    File "magnetostatics.pyx", line 178, in espressomd.magnetostatics.DipolarP3M._tune
    File "magnetostatics.pyx", line 243, in espressomd.magnetostatics.DipolarP3M.python_dp3m_set_tune_params
  TypeError: an integer is required

```
Code to reproduce the issue is enclosed to the zip file. Link to the zip file is https://1drv.ms/u/s!ApqC1jKqCWuWhLwsSI4D9Ka16Le7vQ?e=saFnPy
It contains files for saving and loading checkpoint.

My suggestion that the reason for the issue is indeed in line 243 from magnetostatics.pyx. It should be changed on something like this

```
if hasattr(p_mesh, "__getitem__"):
    mesh = p_mesh[0]
else:
    mesh = p_mesh
```

**Comment by jngrad** [comment]
Thanks for reporting the issue. The bug is present in both the 4.0 and 4.1 lines of ESPResSo.  I can confirm the suggested change fixes the magnetostatics checkpointing mechanism. Unfortunately, there is no planned 4.1.5 release.

It has already been fixed in the upcoming 4.2 release by #3879 (commit 45e38f4bc6910cd7). I've updated the 4.2 release notes accordingly.

**Comment by Vladimir-Z** [comment]
Thank you for answer. We will wait upcoming 4.2 release.
